### ISSUE TITLE
ADAL logout doesn't work

### ISSUE THREAD

**Issue by 1gravity** [OneDrive/onedrive-sdk-android#66]
Logging out of the ADAL authenticator doesn't work (it does for MSA). As a result I'm not able to switch between different OneDrive accounts (my app allows to select different cloud storage providers so they can use OneDrive, Dropbox or others).
To test this I login to OneDrive for Business, then switch to e.g. Dropbox. My app logs out of OneDrive as soon as the user has authenticated with Dropbox.
If I try to switch back to OneDrive, I would expect the disambiguation to kick in again which would allow me to pick another OneDrive account instead of the one I picked in the first place. Since the logout doesn't work, there's no disambiguation and I end of with the same acccount.

Here's the offending piece of code:

```
if (mMSAAuthenticator.getAccountInfo() != null) {
    mLogger.logDebug("Starting logout of MSA account");
    mMSAAuthenticator.logout();
} else if (mADALAuthenticator.getAccountInfo() != null) {
    mLogger.logDebug("Starting logout of ADAL account");
    mADALAuthenticator.logout();
}
```

What happens is that mMSAAuthenticator.getAccountInfo() is always !=null and the mADALAuthenticator.logout() is never executed. The session in the MSA authenticator is always initialized and so the getAccountInfo would always return an object even if it's really just an "empty shell":

```
{
/ /...
    this.session = new LiveConnectSession(this);
}
```

I guess what the code should be doing is something like this:

```
AccountType accountType = getAccountTypeInPreferences();
if (accountType == AccountType.MicrosoftAccount) {
    mLogger.logDebug("Starting logout of MSA account");
    mMSAAuthenticator.logout();
} else if (accountType == AccountType.ActiveDirectory) {
    mLogger.logDebug("Starting logout of ADAL account");
    mADALAuthenticator.logout();
}
```

**Comment by peternied** [comment]
This was fixed in 4f9392e02ffbee8c471c5bdcaf384c47d3200a52 which was included in the 1.1.4 release.

**Comment by 1gravity** [comment]
Sorry my bad
